src/Helper/Cart.php

```diff
@@ -0,0 +1,147 @@
+     * @param RequestInterface  $request
+     * @param array             $quoteFields
+     * @param array             $quoteStockFields
+     */
+    public function __construct(
+        Config $config,
+        SgLoggerInterface $logger,
+        QuoteHelper $quoteHelper,
+        RequestInterface $request,
+        array $quoteFields = [],
+        array $quoteStockFields = []
+    ) {
+        $this->config        = $config;
+        $this->logger        = $logger;
+        $this->quoteHelper   = $quoteHelper;
+        $this->quoteFields   = $quoteFields;
+        $this->request       = $request;
+        $this->isBoltRequest = $this->getBoltRequestFlag($request);
+        parent::__construct($config, $logger, $quoteHelper, $quoteFields, $quoteStockFields);
+    }
+
+    /**
+     * Takes in the allowed cart methods loaded up in the Base's DI
+     *
+     * @return array
+     * @throws \Exception
+     */
+    public function loadSupportedMethods()
+    {
+        $this->quoteHelper->load($this->quoteFields);
+        $fields = $this->loadMethods($this->config->getSupportedFieldsCheckCart());
+        if ($this->isBoltRequest) {
+            $this->quoteHelper->addBoltParentId();
+        }
+        if (!$this->isBoltRequest) {
+            $this->quoteHelper->cleanup();
+        }
+
+        return $fields;
+    }
+
+    /**
+     * Loads the methods of the current class
+     *
+     * @param array $fields
+     *
+     * @return array
+     */
+    private function loadMethods(array $fields)
+    {
+        $methods = [];
+        foreach ($fields as $rawField) {
+            $method = 'get' . SimpleDataObjectConverter::snakeCaseToUpperCamelCase($rawField);
+            $this->logger->debug('Starting method ' . $method);
+            $methods[$rawField] = $this->{$method}();
+            $this->logger->debug('Finished method ' . $method);
+        }
+
+        return $methods;
+    }
+
+    /**
+     * @return mixed
+     */
+    protected function getInternalCartInfo()
+    {
+        $isRequestEmpty = empty($this->request) ? 'yes' : 'no';
+        $requestCart = $this->request->getParam('cart');
+        $internalCartInfo = $requestCart['internal_cart_info'];
+        if ($this->isBoltRequest) {
+            return json_encode([
+                'quote_id' => $this->quoteHelper->getCurrentQuoteId(),
+                'reserved_order_id' => $this->quoteHelper->getReservedOrderId(),
+                'new_cart_used' => 'yes',
+                'is_request_empty' => $isRequestEmpty,
+                'request_internal_cart_info' => $internalCartInfo,
+                'class_is_bolt_request' => $this->isBoltRequest
+            ]);
+        }
+        return '{}';
+    }
+
+    private function getBoltRequestFlag(RequestInterface $request)
+    {
+        $requestCart = $request->getParam('cart');
+
+        if (empty($requestCart)) {
+            return false;
+        }
+        $internalCartInfo = new \stdClass;
```

Comment:
I guess you don't need the stdClass at all. I would recommend renaming the method to isBoltRequest as you only return booleans. Therefore returning (bool)$requestCart['internal_cart_info']['bolt_request'] (only if it exists ;)) or false otherwise. If you want to use something like stdClass, Magento offers you the \Magento\Framework\DataObject class.

Reply:
done